gr-blocks/lib/file_source_impl.cc

```diff
@@ -110,6 +110,16 @@ namespace gr {
 	throw std::runtime_error("can't open file");
       }
 
+			//Check to ensure the file will be consumed according to item size
```

Comment:
See the conversation [here](https://stackoverflow.com/questions/5907162/file-size-lookup-in-c) about using fseek/ftell vs. stat for file size. The performance difference here is negligible as its not inside a loop, bit it is a more direct way to accomplish what you want.